Source/Shaders/Builtin/Functions/discardIfClipped.glsl

```diff
@@ -6,31 +6,48 @@
  *
  * @param {vec4[]} clippingPlanes The array of planes used to clip, defined in eyespace.
  * @param {int} clippingPlanesLength The number of planes in the array of clipping planes.
+ * @param {bool} inclusive
+ * @returns {float} The distance away from a clipped fragment, in eyespace
  */
-void czm_discardIfClipped (vec4[czm_maxClippingPlanes] clippingPlanes, int clippingPlanesLength)
+float czm_discardIfClipped (vec4[czm_maxClippingPlanes] clippingPlanes, int clippingPlanesLength, bool inclusive)
 {
     if (clippingPlanesLength > 0)
     {
-        bool clipped = false;
+        bool clipped = inclusive;
         vec4 position = czm_windowToEyeCoordinates(gl_FragCoord);
         vec3 clipNormal = vec3(0.0);
         vec3 clipPosition = vec3(0.0);
+        float clipAmount = 0.0;
 
         for (int i = 0; i < czm_maxClippingPlanes; ++i)
         {
-            if (i == clippingPlanesLength || clipped)
+            if (i == clippingPlanesLength)
             {
                 break;
             }
 
             clipNormal = clippingPlanes[i].xyz;
             clipPosition = -clippingPlanes[i].w * clipNormal;
-            clipped = dot(clipNormal, (position.xyz - clipPosition)) <= 0.0;
+
+            float amount = dot(clipNormal, (position.xyz - clipPosition));
+            if (amount > clipAmount) {
+                clipAmount = amount;
+            }
+
+            if (inclusive) {
```

Comment:
Don't test this per loop iteration or even in the shader at all - instead, generate a different shader depending on if it should be inclusive or not - they will have the best performance and use the fewest uniforms.

Make sure to recompile the shader - if and only if - the inclusive flag flips.